platform/mbed_wait_api_rtos.cpp

```diff
@@ -23,26 +23,45 @@
 #include "rtos/rtos.h"
 #include "platform/mbed_critical.h"
 #include "platform/mbed_power_mgmt.h"
+#include "platform/mbed_error.h"
 
 void wait(float s)
 {
-    wait_us(s * 1000000.0f);
+    if (s >= 0.01f) {
+        wait_ms(s * 1000.0f);
+    } else {
+        wait_us(s * 1000000.0f);
+    }
 }
 
+/*  The millisec value specifies the number of timer ticks and is therefore an upper bound.
+ *  The exact time delay depends on the actual time elapsed since the last timer tick.
+ *  The actual time delay may be up to one timer tick less.
+ */
 void wait_ms(int ms)
 {
-    wait_us(ms * 1000);
+    if (core_util_are_interrupts_enabled()) {
+        ThisThread::sleep_for(ms);
+    } else {
+        MBED_WARNING(MBED_MAKE_ERROR(MBED_MODULE_PLATFORM, MBED_ERROR_INVALID_OPERATION),
```

Comment:
Does MBED_WARNING actually output a visible string at the moment? I see `MBED_WARNING` calling `mbed_warning` with a string, which it then doesn't use?

My preference for this is to be as harsh as an error, but only on debug builds. I think that's consistent with other nasties we check like RTX errors.

Doing a warning here would likely result in a flood of output or buffer filling, if it wasn't a "once-only" - it could cause extra subtle problems on a normal build, when we're trying to provide a mechanism to catch them.

Reply:
If you don;t have strong opinion on `error`, I can fix `mbed_warning` to add debug print and make this warning once-only. 

Adding `error` will result into `mbed_die` and next instruction will not be executed anyways. So I don;t see a point of calling `wait_us` after `error` unless we add some json config to disable the error.

Reply:
I was suggesting flagging the error for the interrupt case to be something like

    if (in interrupt)
    { 
    #ifdef MBED_DEBUG // or MBED_TRAP_ERRORS_ENABLED ?
       error(blah)
    #else
       wait_us(ms * 1000);
    #endif
    }

At some point in the future, once we think that debug error might have persuaded enough people, we get rid of the interrupt check, just call `ThisThread::sleep_for`, which would itself go pop-bang if called from interrupt.

If it is a warning, once-only is kind of good when the condition is something like this, which is a design error, likely to be repeated throughout every run, rather than an environmental event. But flagging it does mean static data to track the once-only (or rate limiting).

Reply:
Fair enough!! Can use  `MBED_TRAP_ERRORS_ENABLED` since it is set for debug profile and mostly used for similar purpose in RTOS.